plugin/pkg/scheduler/core/extender_test.go

```diff
@@ -112,20 +113,82 @@ type FakeExtender struct {
 	filteredNodes    []*v1.Node
 }
 
+// selectVictimsOnNodeByExtender checks the given nodes->pods map with predicates on extender's side.
+// Returns:
+// 1. More victim pods (if any) amended by preemption phase of extender.
+// 2. Fits or not after preemption phase on extender's side.
+func (f *FakeExtender) selectVictimsOnNodeByExtender(
+	pod *v1.Pod,
+	node *v1.Node,
+	originVictims *schedulerapi.Victims,
+	pdbs []*policy.PodDisruptionBudget,
+) ([]*v1.Pod, int, bool) {
+	if fits, _ := f.runPredicate(pod, node); !fits {
+		return nil, 0, false
+	}
+
+	victims := []*v1.Pod{}
+	numViolatingVictim := 0
+
+	return victims, numViolatingVictim, true
+}
+
+func (f *FakeExtender) SupportsPreemption() bool {
+	return true
+}
+
+func (f *FakeExtender) ProcessPreemption(
```

Comment:
Adding this function is not going to exercise preemption in extenders, unless I am missing something. We need a test that exercises the new logic.

Reply:
Hope I am understanding it rightly, I assume this should be called during `generic_scheduler_test.go#TestPreempt()`.

Reply:
My understanding is that `TestSchedulerExtender` creates a scheduler with several extenders and then calls `DoTestPodScheduling` to try scheduling one pod. Scheduling the pod in an otherwise empty cluster is not going to trigger preemption. That's why I think adding this function is not going to help by itself. You need to also ensure that preemption is triggered. To do so, you should enable the flag and then create pods with lower priority that fill the existing nodes and then create another pod that wouldn't fit anywhere unless some pods are preempted.

Reply:
Oh, I see, seems I missed the integration test part, will amend this work!

Reply:
@bsalamat Emm, just revisited the integration test, while as we once discussed, if we can not pass `nodeInfo` to extender, mocking `create pods with lower priority that fill the existing nodes` is not practical since even a simple preemption in extender still requires `nodeInfo.Pods` at least.

One option is maybe we can pass a simplified `nodeInfo` to extender (only with resources related fields in pods)?
Otherwise, we have to only test extender against a very simple case which will only remove some candidate nodes, no preemption will happen. (same as the unit test I added today)

Reply:
This might be a problem for real extenders as well. Our current interface sends a map of node name to victims to the extender. How about sending a map of node name to pods for the same set of nodes as the victims map to the extender as well?

Reply:
If only extending the interface of `ProceesPreemption()`, passing a list of victims pods should be fine. While we may still need to send several resource information (requestedResource, allocatableResource) of those node for decision making, luckily the size of those info should be fixed instead of huge data structure. What do you think?

Reply:
Hmm... basically we get closer to sending node info to the extenders, because the extenders may care about other properties of nodes than just resources. In that case, wouldn't it be better to use the same Filter verb? Here is a sketch of the algorithm that I have in mind:
1. Scheduler runs preemption and find a set of nodes and victims on each one.
2. Scheduler removes the victims from NodeInfo of those nodes.
3. Scheduler sends the preemptor Pod, nodes (with their NodeInfo) to extenders with Filter verb.
4. Scheduler receives a filtered list of nodes from the extenders.
5. Scheduler applies PDB and other scoring to the filtered list of nodes to pick one node for preemption.